GUI/src/javafxgui/controller/GUIAppController.java

```diff
@@ -41,37 +42,44 @@ public void displayResult() {
     }
 
     @Override
-    public void displayPlayAgain() {
+    public int displayPlayAgainOption() {
         if (foundWinOrDraw()) {
             guiView.makePlayAgainVisible();
         }
+        return 0; //Potential LSP violation as return value isn't used in GUI
```

Comment:
Good catch. This tells us that perhaps the abstraction probably isn't quite right; might need to apply ISP in order to make it not violate LSP, but there's a cost to that. Would like to have you just have a think about this for your IPM.